services/requester/requester.go

```diff
@@ -631,14 +631,18 @@ func (e *EVM) buildTransaction(
 
 	for _, arg := range args {
 		if err := flowTx.AddArgument(arg); err != nil {
+			accKey.Done()
```

Comment:
Instead of adding this `accKey.Done()` call, we could simply move down this piece of code:
```go
accKey, err := e.keystore.Take()
if err != nil {
	return nil, err
}
```
Ideally, just before calling `accKey.SetProposerPayerAndSign`.

Reply:
good call